Output checking measures

LFISHER7 opened the pull request:
Addresses #116 and #30

ccunningham101 commented:
Will take a look at this later this week or early next week, but general question about having a separate measure file for checking. When I think about output checking underlying data for tables and figures, I would prefer that I can follow the direct code path that generates the tables and figures from the data I am checking. 
In theory it would be possible to modify the `measure_df_for_checking.csv` file so that it doesn't match what is in the tables and figures.

LFISHER7 commented:
I've updated this so the measures file to be checked is now producing in `plot_measures.py`, just before it is used for plotting. This should make it easier to confirm it aligns with what is being plotted

ccunningham101 commented on interactive_templates/templates/v2/analysis/measures.py:
```diff
@@ -133,6 +133,7 @@ def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument("--breakdowns", action="append", default=[], required=False)
     parser.add_argument("--input-dir", type=str, required=True)
+    parser.add_argument("--output-dir", type=str, required=True)
```
This doesn't actually appear to be used? Starting at lines 187 the output is currently saved to args.input_dir

ccunningham101 commented on interactive_templates/templates/v2/analysis/plot_measures.py:
```diff
@@ -21,6 +22,15 @@ def main():
+    # subset of the measures file that is used for plotting in this script
+    df = df.loc[~(df["group"].isin(["event_1_code", "event_2_code", "practice"])), :]
+
+    Path(f"{args.output_dir}/for_checking").mkdir(parents=True, exist_ok=True)
```
Not a necessary change, but if you are using the pathlib library, you don't have to use format strings
So you could specify from the start that the argparse output dir argument is type=Path
Then do 
`(args.output_dir / "for_checking").mkdir(parents=True, exist_ok=True)`
and then pandas.read_csv can accept a path object rather than a string

ccunningham101 commented on interactive_templates/templates/v2/analysis/plot_measures.py:
```diff
@@ -21,6 +22,15 @@ def main():
     df = pd.read_csv(
         f"{ args.output_dir }/joined/measure_all.csv", parse_dates=["date"]
```
Might it be more flexible to pass as an argument the path to the input file separate from the output_dir? What if measure_all does not share the same path root as where you want to put the output?

ccunningham101 reviewed (changes requested):
Could either remove output-dir argument in measures.py or switch to using that instead of input-dir when outputting files
Otherwise everything else looks good

LFISHER7 commented:
Thanks @ccunningham101, i've added separate input_dir and output_dir args for extra flexibility and converted them to use `Path` to avoid using f strings

ccunningham101 commented on interactive_templates/templates/v2/analysis/plot_measures.py:
```diff
@@ -40,19 +42,19 @@ def main():
             plot_measures(
                 df_subset,
-                filename=f"{ args.output_dir }/plot_measures_{breakdown}",
+                filename=args.output_dir / "plot_measures_{breakdown}",
```
missing a f' here to resolve {breakdown}

ccunningham101 reviewed (approved):
Just the fix to imd {breakdown} and then otherwise looks good!
